<launch>

  <!-- args corresponding to environment variables -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav1)" />
  <arg name="RUN_TYPE" default="$(optenv RUN_TYPE simulation)" />

  <arg name="custom_config" default="" />

  <arg if="$(eval arg('RUN_TYPE') == 'simulation')" name="simulation" default="true" />
  <arg if="$(eval arg('RUN_TYPE') == 'realworld')" name="simulation" default="false" />

  <arg name="debug" default="false" />
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

    <!-- other args -->
  <arg name="standalone" default="true" />

  <arg     if="$(arg standalone)" name="nodelet" value="standalone" />
  <arg unless="$(arg standalone)" name="nodelet" value="load" />
  <arg     if="$(arg standalone)" name="nodelet_manager" value="" />
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg UAV_NAME)_vision_nodelet_manager" />

  <group ns="$(arg UAV_NAME)">

    <node name="automatic_start" pkg="nodelet" type="nodelet" args="$(arg nodelet) mrs_uav_autostart/AutomaticStart $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <param name="config_private" value="$(find mrs_uav_autostart)/config/private/automatic_start.yaml" />
      <param name="config_public" value="$(find mrs_uav_autostart)/config/public/automatic_start.yaml" />

      <param if="$(eval arg('custom_config') == '')" name="custom_config" value="" />
      <param if="$(eval not arg('custom_config') == '' and arg('custom_config')[0] == '/')" name="custom_config" value="$(arg custom_config)" />
      <param if="$(eval not arg('custom_config') == '' and not arg('custom_config')[0] == '/')" name="custom_config" value="$(env PWD)/$(arg custom_config)" />

      <param name="uav_name" value="$(arg UAV_NAME)" />
      <param name="simulation" value="$(arg simulation)" />

        <!-- Subscribers -->
      <remap from="~hw_api_status_in" to="hw_api/status" />
      <remap from="~control_manager_diagnostics_in" to="control_manager/diagnostics" />
      <remap from="~uav_manager_diagnostics_in" to="uav_manager/diagnostics" />
      <remap from="~gazebo_spawner_diagnostics_in" to="/mrs_drone_spawner/diagnostics" />
      <remap from="~uav_state_in" to="estimation_manager/diagnostics" />

        <!-- Publishers -->
      <remap from="~can_takeoff_out" to="~can_takeoff" />

        <!-- services -->
      <remap from="~takeoff_out" to="uav_manager/takeoff" />
      <remap from="~eland_out" to="control_manager/eland" />
      <remap from="~toggle_control_output_out" to="control_manager/toggle_output" />
      <remap from="~arm_out" to="hw_api/arming" />
      <remap from="~validate_reference_out" to="control_manager/validate_reference_2d" />

    </node>

  </group>

</launch>
